pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_builtin_function failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_builtin_function [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14560352?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14560352?buildTab=artifacts#/) on master @ [215ece44cfba2ab950570f9d03cf29599e1ad088](https://github.com/cockroachdb/cockroach/commits/215ece44cfba2ab950570f9d03cf29599e1ad088):


```
=== RUN   TestTenantLogic_builtin_function
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/5b0e6ab5264dcff5c30a4ba875ec154a/logTestTenantLogic_builtin_function2104150777
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:4322: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b0e6ab5264dcff5c30a4ba875ec154a/logTestTenantLogic_builtin_function2104150777
--- FAIL: TestTenantLogic_builtin_function (14.48s)
=== RUN   TestTenantLogic_builtin_function/check_consistency
    logic.go:2975: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5544/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/builtin_function:4040: SELECT status, regexp_replace(detail, '[0-9]+', '', 'g')
        FROM crdb_internal.check_consistency(true, crdb_internal.tenant_span()[1], crdb_internal.tenant_span()[2])
        ORDER BY range_id
        LIMIT 1
        expected:
            RANGE_CONSISTENT  stats: {ContainsEstimates: LastUpdateNanos: LockAge: GCBytesAge: LiveBytes: LiveCount: KeyBytes: KeyCount: ValBytes: ValCount: IntentBytes: IntentCount: LockBytes: LockCount: RangeKeyCount: RangeKeyBytes: RangeValCount: RangeValBytes: SysBytes: SysCount: AbortSpanBytes:}
            
        but found (query options: "") :
            RANGE_CONSISTENT  stats: {ContainsEstimates: LastUpdateNanos: LockAge:- GCBytesAge: LiveBytes: LiveCount: KeyBytes: KeyCount: ValBytes: ValCount: IntentBytes:- IntentCount:- LockBytes: LockCount:- RangeKeyCount: RangeKeyBytes: RangeValCount: RangeValBytes: SysBytes: SysCount: AbortSpanBytes:}
            
            
    logic.go:2212: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5544/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/builtin_function:4048: too many errors encountered, skipping the rest of the input
[06:18:33] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5544/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/builtin_function with config 3node-tenant: 705 tests, 2 failures
[06:18:34] --- total progress: 705 statements
--- total: 705 tests, 2 failures
    --- FAIL: TestTenantLogic_builtin_function/check_consistency (0.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_builtin_function.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37071

--- a/pkg/sql/logictest/testdata/logic_test/builtin_function
+++ b/pkg/sql/logictest/testdata/logic_test/builtin_function
@@ -4037,7 +4037,7 @@ statement error start key must be less than end key
 SELECT crdb_internal.check_consistency(true, crdb_internal.tenant_span()[2], crdb_internal.tenant_span()[2])
 
 query TT
-SELECT status, regexp_replace(detail, '[0-9]+', '', 'g')
+SELECT status, regexp_replace(detail, '[0-9-]+', '', 'g')
 FROM crdb_internal.check_consistency(true, crdb_internal.tenant_span()[1], crdb_internal.tenant_span()[2])
 ORDER BY range_id
 LIMIT 1

